src/script_compiler.rs

```diff
@@ -85,30 +97,44 @@ enum BufferPolicy {
 }
 
 impl Source {
-  pub fn new(source_string: Local<String>, origin: &ScriptOrigin) -> Self {
+  pub fn new<'a, 'b, T>(source_string: Local<String>, origin: T) -> Self
+  where
+    'a: 'b,
+    T: Into<Option<&'b ScriptOrigin<'a>>>,
```

Comment:
I guess you're doing this to remain backwards compatible, but personally I prefer an explicit `Option<&ScriptOrigin>` if this were the only reason. Small changes in the API like this are acceptable  at this early stage in the project.

Reply:
Yeah it's a trick to maintain backward compatibility. Removed.